docs/sections/cn_workflows.md

```diff
@@ -0,0 +1,415 @@
+
+```yaml
+version: '1'
+steps:
+- uses: "sh"
+  runs: ["ls", "-la"]
+
+- uses: "sh"
+  runs: "./path/to/my/script.sh"
+  args: ["some", "args", "to", "the", "script"]
+```
+
+In the first step above, the `ls -la` command is executed on the 
+workspace folder (see ["The workspace"](#the-workspace) section). The 
+second one shows how to execute a script. Note that the command or 
+script specified in the `runs` attribute are NOT executed in a shell. 
+If you need a shell, you have to explicitly invoke one, for example:
+
+```yaml
+version: '1'
+steps:
+- uses: sh
+  runs: [bash, -c, 'sleep 10 && true && exit 0']
+```
+
+The obvious downside of running a step on the host is that, depending 
+on the command being executed, the workflow might not be portable.
+
+### Custom engine configuration
+
+Other than bind-mounting the `/workspace` folder, Popper runs 
+containers with any default configuration provided by the underlying 
+engine. However, a `--conf` flag is provided by the `popper run` 
+command to specify custom options for the underlying engine in 
+question (see [here][engconf] for more).
+
+[engconf]: /cli_features#customizing-container-engine-behavior
+
+## Resource Managers
+
+Popper can execute steps in a workflow through other resource managers
+like SLURM besides the host machine. The resource manager can be specified 
+either through the `--resource-manager/-r` option or through the config file.
+If neither of them are provided, the steps are run in the host machine 
+by default. 
+
+### SLURM
+
+Popper workflows can run on [HPC](https://en.wikipedia.org/wiki/HPC) (Multi-Node environments) 
+using [Slurm](https://slurm.schedmd.com/overview.html) as the underlying resource manager to distribute the execution of a step to
+several nodes. You can get started with running Popper workflows through Slurm by following the example below.
+
+Let's consider a workflow `sample.yml` like the one shown below.
+```yaml
+version: '1'
+steps:
+- id: one
+  uses: docker://alpine:3.9
+  args: echo hello-world
+
+- id: two
+  uses: popperized/bin/sh@master
+  args: ls -l
+```
+
+To run all the steps of the workflow through slurm resource manager,
+use the `--resource-manager` or `-r` option of the `popper run` subcommand to specify the resource manager.
+
+```bash
+popper run -f sample.yml -r slurm
+```
+
+To have more finer control on which steps to run through slurm resource manager,
+the specifications can be provided through the config file as shown below.
+
+We create a config file called `config.yml` with the following contents.
+
+```yaml
+engine:
+  name: docker
+  options:
+    privileged: True
+    hostname: example.local
+
+resource_manager:
+  name: slurm
+  options:
+    two:
+      nodes: 2
+```
+
+Now, we execute `popper run` with this config file as follows:
+
+```bash
+popper run -f sample.yml -c config.yml
+```
+
+This runs the step `one` locally in the host and step `two` through slurm on 2 nodes.
+
+## Kubernetes
```

Comment:
@ivotron I'm trying to ensure this is stored somewhere safe. So when @JayjeetAtGithub finishes with the Kubernetes runner we can be sure there's a documentation for it. Also, what's the hold up with the Kubernetes resource manager?